# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/PeerConnectionMedia.cpp
# Commit: 3ca24e761cc3
# Full Hash: 3ca24e761cc31bbb71c659661874a995ef3d3db1
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 6
# Description:
#   Bug 1654112 - Remove WebrtcGmpPCHandleSetter and plumb pc handles explicitly. r=bwc
#   
#   With the libwebrtc update codecs will be created off-main.
#   WebrtcGmpPCHandleSetter not only adds a lot of indirection, it is also
#   main-thread-only. With this patch we can create gmp codecs on any thread.
# ==============================================================================

diff -r 1913c332695e -r 3ca24e761cc3 dom/media/webrtc/jsapi/PeerConnectionMedia.cpp
--- a/dom/media/webrtc/jsapi/PeerConnectionMedia.cpp	Wed Feb 10 09:58:11 2021 +0100
+++ b/dom/media/webrtc/jsapi/PeerConnectionMedia.cpp	Wed Feb 10 10:34:14 2021 +0100
@@ -299,12 +299,6 @@
 }
 
 nsresult PeerConnectionMedia::UpdateMediaPipelines() {
-  // The GMP code is all the way on the other side of webrtc.org, and it is not
-  // feasible to plumb error information all the way back. So, we set up a
-  // handle to the PC (for the duration of this call) in a global variable.
-  // This allows the GMP code to report errors to the PC.
-  WebrtcGmpPCHandleSetter setter(mParentHandle);
-
   for (RefPtr<TransceiverImpl>& transceiver : mTransceivers) {
     transceiver->ResetSync();
   }